programs/ssl/ssl_client2.c

```diff
@@ -2591,7 +2739,39 @@ int main( int argc, char *argv[] )
         }
         mbedtls_printf("\n");
     }
-#endif
+
+#if defined( MBEDTLS_SSL_DTLS_SRTP )
+    else if( opt.use_srtp != 0  )
+    {
+        size_t j = 0;
+
+        if( ( ret = mbedtls_ssl_tls_prf( dtls_srtp_keying.tls_prf_type,
```

Comment:
I don't think we've checked at this point that DTLS SRTP has actually been negotiated, or did we?

Reply:
I added a check : fetch the negotiated protection profile. As newly explained in the doc, it is available as soon as the server Hello as been parsed.